test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTxnWithFailoverTest.java

```diff
@@ -0,0 +1,444 @@
+            });
+
+            //perform the scaling operations
+            performFailoverTest();
+
+            //set the stop write flag to true
+            log.info("Stop write flag status {}", stopWriteFlag);
+            stopWriteFlag.set(true);
+
+            //wait for writers completion
+            log.info("Wait for writers execution to complete");
+            FutureHelpers.allOf(writerFutureList).get();
+
+            //set the stop read flag to true
+            log.info("Stop read flag status {}", stopReadFlag);
+            stopReadFlag.set(true);
+
+            //wait for readers completion
+            log.info("Wait for readers execution to complete");
+            FutureHelpers.allOf(readerFutureList).get();
+
+            log.info("All writers have stopped. Setting Stop_Read_Flag. Event Written Count:{}, Event Read " +
+                    "Count: {}", eventWriteCount.get(), eventsReadFromPravega.size());
+            assertEquals(eventWriteCount.get(), eventsReadFromPravega.size());
+            assertEquals(eventWriteCount.get(), new TreeSet<>(eventsReadFromPravega).size()); //check unique events.
+
+            //close all the writers
+            log.info("Closing writers");
+            writerList.forEach(writer -> writer.close());
+            //close all readers
+            log.info("Closing readers");
+            readerList.forEach(reader -> reader.close());
+            //delete readergroup
+            log.info("Deleting readergroup {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+        //seal the stream
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(scope, STREAM_NAME);
+        log.info("Sealing stream {}", STREAM_NAME);
+        assertTrue(sealStreamStatus.get());
+        //delete the stream
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(scope, STREAM_NAME);
+        log.info("Deleting stream {}", STREAM_NAME);
+        assertTrue(deleteStreamStatus.get());
+
+        //delete the scope
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(scope);
+        log.info("Deleting scope {}", scope);
+        assertTrue(deleteScopeStatus.get());
+        log.info("Test {} succeeds ", "MultiReaderWriterTxnWithFailOver");
+    }
+
+    private void performFailoverTest() throws InterruptedException {
+
+        long currentWriteCount1;
+        long currentReadCount1;
+
+        log.info("Test with 2 controller, SSS instances running and without a failover scenario");
+
+        currentWriteCount1 = eventWriteCount.get();
+        currentReadCount1 = eventReadCount.get();
+
+        log.info("Read count without any failover {}", currentReadCount1);
+        log.info("Write count without any failover {}", currentWriteCount1);
+
+        int sleepTime;
+
+        //check reads and writes after some random time
+        sleepTime = new Random().nextInt(50000) + 3000;
```

Comment:
what is this delay giving us.. a delay after failover is meaningful but a delay before failure is not going to serve any purpose IMO